torch/csrc/lazy/core/shape_inference.cpp

```diff
@@ -508,7 +508,7 @@ std::vector<Shape> compute_shape_cat(at::TensorList tensors, int64_t dim) {
       tensors[0].sizes().begin(), tensors[0].sizes().end());
 
   dim = at::maybe_wrap_dim(dim, tensors);
-  size_t extended_dim_shape = 0;
```

Comment:
This change is required because of this failure:
https://github.com/pytorch/pytorch/actions/runs/5113544004/jobs/9192901211

Here is the log:
```
var/lib/jenkins/workspace/third_party/ideep/include -isystem /var/lib/jenkins/workspace/third_party/ideep/mkl-dnn/include -isystem /var/lib/jenkins/cpp-build/caffe2/build/include -D_GLIBCXX_USE_CXX11_ABI=1 -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -DUSE_KINETO -DLIBKINETO_NOROCTRACER -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wall -Wextra -Werror=return-type -Werror=non-virtual-dtor -Werror=bool-operation -Wnarrowing -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=old-style-cast -Wno-invalid-partial-specialization -Wno-unused-private-field -Wno-aligned-allocation-unavailable -Wno-missing-braces -fdiagnostics-color=always -faligned-new -Werror -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow -DHAVE_AVX512_CPU_DEFINITION -DHAVE_AVX2_CPU_DEFINITION -g -fno-omit-frame-pointer -O0 -fPIC -DCAFFE2_USE_GLOO -DTH_HAVE_THREAD -Wall -Wextra -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-missing-field-initializers -Wno-unknown-pragmas -Wno-type-limits -Wno-array-bounds -Wno-strict-overflow -Wno-strict-aliasing -Wno-missing-braces -Wno-maybe-uninitialized -fvisibility=hidden -pthread -DASMJIT_STATIC -fopenmp -std=gnu++17 -MD -MT caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/runtime/register_prim_ops.cpp.o -MF CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/runtime/register_prim_ops.cpp.o.d -o CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/runtime/register_prim_ops.cpp.o -c /var/lib/jenkins/workspace/torch/csrc/jit/runtime/register_prim_ops.cpp
2023-05-29T17:31:21.0985620Z In file included from [01m[K/var/lib/jenkins/workspace/c10/util/Exception.h:4[m[K,
2023-05-29T17:31:21.0986403Z                  from [01m[K/var/lib/jenkins/workspace/c10/core/Device.h:5[m[K,
2023-05-29T17:31:21.0987189Z                  from [01m[K/var/lib/jenkins/cpp-build/caffe2/build/aten/src/ATen/core/TensorBody.h:11[m[K,
2023-05-29T17:31:21.0990132Z                  from [01m[K/var/lib/jenkins/workspace/aten/src/ATen/core/Tensor.h:3[m[K,
2023-05-29T17:31:21.0991227Z                  from [01m[K/var/lib/jenkins/workspace/aten/src/ATen/Tensor.h:3[m[K,
2023-05-29T17:31:21.0992332Z                  from [01m[K/var/lib/jenkins/workspace/torch/csrc/lazy/core/shape_inference.h:3[m[K,
2023-05-29T17:31:21.0993570Z                  from [01m[K/var/lib/jenkins/workspace/torch/csrc/lazy/core/shape_inference.cpp:50[m[K:
2023-05-29T17:31:21.0995246Z [01m[K/var/lib/jenkins/workspace/torch/csrc/lazy/core/shape_inference.cpp:[m[K In function ‘[01m[Kstd::vector<torch::lazy::Shape> torch::lazy::compute_shape_cat(at::TensorList, int64_t)[m[K’:
2023-05-29T17:31:21.1063780Z [01m[K/var/lib/jenkins/workspace/torch/csrc/lazy/core/shape_inference.cpp:517:26:[m[K [01;31m[Kerror: [m[Kcomparison of integer expressions of different signedness: ‘[01m[Ktorch::lazy::size_t[m[K’ {aka ‘[01m[Klong unsigned int[m[K’} and ‘[01m[Klong int[m[K’ [[01;31m[K-Werror=sign-compare[m[K]
2023-05-29T17:31:21.1065291Z   517 |       [01;31m[Kextended_dim_shape <= std::numeric_limits<int64_t>::max()[m[K,
2023-05-29T17:31:21.1065924Z       |       [01;31m[K~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
2023-05-29T17:31:21.1066759Z [01m[K/var/lib/jenkins/workspace/c10/macros/Macros.h:200:64:[m[K [01;36m[Knote: [m[Kin definition of macro ‘[01m[KC10_UNLIKELY[m[K’
2023-05-29T17:31:21.1067627Z   200 | #define C10_UNLIKELY(expr) (__builtin_expect(static_cast<bool>([01;36m[Kexpr[m[K), 0))
2023-05-29T17:31:21.1068289Z       |                                                                [01;36m[K^~~~[m[K
2023-05-29T17:31:21.1069115Z [01m[K/var/lib/jenkins/workspace/c10/util/Exception.h:503:7:[m[K [01;36m[Knote: [m[Kin expansion of macro ‘[01m[KC10_UNLIKELY_OR_CONST[m[K’
2023-05-29T17:31:21.1069898Z   503 |   if ([01;36m[KC10_UNLIKELY_OR_CONST[m[K(!(cond))) {            \
2023-05-29T17:31:21.1074588Z       |       [01;36m[K^~~~~~~~~~~~~~~~~~~~~[m[K
2023-05-29T17:31:21.1075498Z [01m[K/var/lib/jenkins/workspace/torch/csrc/lazy/core/shape_inference.cpp:516:3:[m[K [01;36m[Knote: [m[Kin expansion of macro ‘[01m[KTORCH_CHECK[m[K’
2023-05-29T17:31:21.1076197Z   516 |   [01;36m[KTORCH_CHECK[m[K(
2023-05-29T17:31:21.1076643Z       |   [01;36m[K^~~~~~~~~~~[m[K
2023-05-29T17:31:21.1077110Z At global scope:
2023-05-29T17:31:21.1078009Z [01m[Kcc1plus:[m[K [01;31m[Kerror: [m[Kunrecognized command line option ‘[01m[K-Wno-aligned-allocation-unavailable[m[K’ [[01;31m[K-Werror[m[K]
2023-05-29T17:31:21.1079109Z [01m[Kcc1plus:[m[K [01;31m[Kerror: [m[Kunrecognized command line option ‘[01m[K-Wno-unused-private-field[m[K’ [[01;31m[K-Werror[m[K]
2023-05-29T17:31:21.1080227Z [01m[Kcc1plus:[m[K [01;31m[Kerror: [m[Kunrecognized command line option ‘[01m[K-Wno-invalid-partial-specialization[m[K’ [[01;31m[K-Werror[m[K]
2023-05-29T17:31:21.1080892Z cc1plus: all warnings being treated as errors
2023-05-29T17:31:21.1081830Z caffe2/CMakeFiles/torch_cpu.dir/build.make:11900: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/lazy/core/shape_inference.cpp.o' failed
2023-05-29T17:31:21.1082612Z make[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/lazy/core/shape_inference.cpp.o] Error 1
2023-05-29T17:31:21.1083173Z make[2]: *** Waiting for unfinished jobs....
2023-05-29T17:31:21.1083952Z [ 83%] [32mBuilding CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp.o[0m
```

Reply:
Yes, but it renders following check tautological  https://github.com/pytorch/pytorch/blob/e71ab214226af1f9dbded944e939c6447e0e8f09/torch/csrc/lazy/core/shape_inference.cpp#L516-L517

So instead of changing the time of extended_dim_shape, please add a cast to `std::numeric_limits`